gtsam/discrete/DecisionTree-inl.h

```diff
@@ -19,22 +19,24 @@
 
 #pragma once
 
-#include <gtsam/discrete/DecisionTree.h>
 #include <gtsam/base/Testable.h>
+#include <gtsam/base/VectorSpace.h>
```

Comment:
Why? Could we make DecisionTree and ADT completely independent of GTSAM? They are nearly so: an elegant, minimal data structure. Maybe even change the namespace. (We are in 4.2 now).

Reply:
Yup, this is because of SFINAE and using the `traits::Equals`. It was also causing the `dot` collision, so I am more than glad to get rid of this.

Reply:
Resolved :D